<launch>

    <arg name="swarm_num" default="3"/>
    <arg name="autoload" default="false"/>
    <arg name="uav_control_start" default=""/>
    <arg name="close_uav_control" default=""/>
    <arg name="swarm_control_start" default=""/>
    <arg name="close_swarm_control" default=""/>
    <arg name="ground_stationt_ip" default="127.0.0.1"/>
    <arg name="uav_id" default="1"/>
    <arg name="is_simulation" default="1"/>
    <arg name="swarm_data_update_timeout" default="5"/>

    <node pkg="prometheus_communication_bridge" type="communication_bridge" name="communication_bridge" output="screen">
        <!--组播ip-->
        <param name="multicast_udp_ip" value="224.0.0.88"/>
	    <!--ego planner-->
        <param name="next_drone_ip" value="127.0.0.1"/>
	    <param name="broadcast_ip" value="127.0.0.255"/>
        <!-- 地面站ip -->
        <param name="ground_stationt_ip" value="$(arg ground_stationt_ip)"/>
        <param name="udp_port" value="8889"/>
        <param name="tcp_port" value="55555"/>
        <param name="rviz_port" value="8890"/>
        <param name="ROBOT_ID" value="$(arg uav_id)"/>
        <param name="try_connect_num" value="5"/>
        <param name="tcp_heartbeat_port" value="55556"/>
        <!--是否仿真模式  1为是  0为否-->
        <param name="is_simulation" value="$(arg is_simulation)"/>
        <!--集群数量，如果不是集群模式值为0-->
        <param name="swarm_num" value="$(arg swarm_num)"/>
        <!-- 集群模式下，数据超时情况，多久判断其失联且回传地面站 -->
        <param name="swarm_data_update_timeout" value="$(arg swarm_data_update_timeout)"/>

        <param name="drone_id" value="1"/>
        <param name="next_drone_ip" value="127.0.0.1"/>
        <param name="broadcast_ip" value="127.0.0.255"/>

        <param name="trajectory_flag" value="false"/>
        <param name="trajectory_mode" value="0"/>
        <param name="trajectory_time" value="0"/>

        <!-- 是否自启 -->
        <param name="autoload" value="$(arg autoload)"/>
        <param name="uav_control_start" value="$(arg uav_control_start)"/>
        <param name="close_uav_control" value="$(arg close_uav_control)"/>
        <param name="swarm_control_start" value="$(arg swarm_control_start)"/>
        <param name="close_swarm_control" value="$(arg close_swarm_control)"/>

        <rosparam command="load" file="$(find prometheus_communication_bridge)/../uav_control/launch/uav_control_outdoor.yaml" />
    </node>
</launch>
